assets/js/atomic/components/product/button/index.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import apiFetch from '@wordpress/api-fetch';
+import { __, sprintf } from '@wordpress/i18n';
+import { Component } from 'react';
+import { addQueryArgs } from '@wordpress/url';
+
+class ProductButton extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		product: PropTypes.object.isRequired,
+	};
+
+	state = {
+		addedToCart: false,
+		addingToCart: false,
+		cartQuantity: null,
+	};
+
+	onAddToCart = () => {
+		const { product } = this.props;
+
+		this.setState( { addingToCart: true } );
+
+		return apiFetch( {
+			method: 'POST',
+			path: '/wc/blocks/cart/add',
+			data: {
+				product_id: product.id,
+				quantity: 1,
+			},
+			cache: 'no-store',
+		} )
+			.then( ( response ) => {
+				const newQuantity = response.quantity;
+
+				this.setState( {
+					addedToCart: true,
+					addingToCart: false,
+					cartQuantity: newQuantity,
+				} );
+			} )
+			.catch( ( response ) => {
+				if ( response.code ) {
+					return ( document.location.href = addQueryArgs(
+						product.permalink,
+						{ wc_error: response.message }
+					) );
+				}
+
+				document.location.href = product.permalink;
+			} );
+	};
+
+	getButtonText = () => {
+		const { product } = this.props;
+		const { cartQuantity } = this.state;
+
+		if ( Number.isFinite( cartQuantity ) ) {
+			return sprintf(
+				__( '%d in cart', 'woo-gutenberg-products-block' ),
```

Comment:
This would need to be a numbered text: `_n( '%d in cart', '%d in cart', cartQuantity, 'woo-gutenberg-products-block' )`

Reply:
https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1413